lib/origen_testers/igxl_based_tester/base/edgesets.rb

```diff
@@ -0,0 +1,53 @@
+module OrigenTesters
+  module IGXLBasedTester
+    class Base
+      class Edgesets
+        include ::OrigenTesters::Generator
+
+        attr_accessor :es
+
+        OUTPUT_PREFIX = 'ES_'
+        OUTPUT_POSTFIX = ''
+
+        def initialize # :nodoc:
+          @es = {}
+        end
+
+        def add(esname, pin, edge, options = {})
+          esname = esname.to_sym unless esname.is_a? Symbol
+          pin = pin.to_sym unless pin.is_a? Symbol
+          @es.key?(esname) ? @es[esname].add_edge(pin, edge) : @es[esname] = platform::Edgeset.new(esname, pin, edge, options)
+          @es
+        end
+
+        def finalize(options = {})
+        end
+
+        # Equality check to compare full contents of edge object
+        def edges_eql?(edge1, edge2)
+          edge1.d_src == edge2.d_src && edge1.d_fmt == edge2.d_fmt &&
+            edge1.d0_edge == edge2.d0_edge && edge1.d1_edge == edge2.d1_edge &&
+            edge1.d2_edge == edge2.d2_edge && edge1.d3_edge == edge2.d3_edge &&
+            edge1.c_mode == edge2.c_mode &&
+            edge1.c1_edge == edge2.c1_edge && edge1.c2_edge == edge2.c2_edge &&
+            edge1.t_res == edge2.t_res && edge1.clk_per == edge2.clk_per
+        end
```

Comment:
Thanks for clarifying @ginty I’m with you now…

Will redefine the == for  edge and level objects

From: Stephen McGinty [mailto:notifications@github.com]
Sent: Friday, March 18, 2016 11:21 AM
To: Origen-SDK/origen_testers <origen_testers@noreply.github.com>
Cc: Aaron Burgmeier <aaron.burgmeier@nxp.com>
Subject: Re: [origen_testers] Support for Edgeset/Timeset sheet generation (#8)


In lib/origen_testers/igxl_based_tester/base/edgesets.rb<https://github.com/Origen-SDK/origen_testers/pull/8#discussion_r56682135>:

> +          @es.key?(esname) ? @es[esname].add_edge(pin, edge) : @es[esname] = platform::Edgeset.new(esname, pin, edge, options)

> +          @es

> +        end

> +

> +        def finalize(options = {})

> +        end

> +

> +        # Equality check to compare full contents of edge object

> +        def edges_eql?(edge1, edge2)

> +          edge1.d_src == edge2.d_src && edge1.d_fmt == edge2.d_fmt &&

> +            edge1.d0_edge == edge2.d0_edge && edge1.d1_edge == edge2.d1_edge &&

> +            edge1.d2_edge == edge2.d2_edge && edge1.d3_edge == edge2.d3_edge &&

> +            edge1.c_mode == edge2.c_mode &&

> +            edge1.c1_edge == edge2.c1_edge && edge1.c2_edge == edge2.c2_edge &&

> +            edge1.t_res == edge2.t_res && edge1.clk_per == edge2.clk_per

> +        end

Don't think we are talking about the same thing here @aaronburg<https://github.com/aaronburg>:

class Edge

  attr_accessor :d0, :d1



  def initialize(d0, d1)

    @d0 = d0

    @d1 = d1

  end



  def ==(edge)

    if edge.is_a?(Edge)

      d0 == edge.d0 && d1 == edge.d1

    else

      super

    end

  end

end



e1 = Edge.new(1, 2)

e2 = Edge.new(1, 2)

e3 = Edge.new(1, 3)



e1 == e2   # => true

e1 == e3   # => false

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Origen-SDK/origen_testers/pull/8/files/8ffdad474d2cf9147fcfe3b93a662d2cc1d1461e#r56682135>